src/manager.rs

```diff
@@ -147,14 +181,9 @@ where
         let rng = rand::rngs::OsRng::default();
         let password: String = rng.sample_iter(&Alphanumeric).take(24).collect();
 
-        let cfg: ServiceConfiguration = signal_servers.into();
+        let mut push_service = self.push_service()?;
```

Comment:
this is not gonna work, because we aren't registered yet: you should initialize a `PushService` with the provided `signal_servers` instead of getting them from the state.

Reply:
ProvisioningManager now override the auth. This was merged.

Reply:
yes, but `fn push_service()` now inspects the state and if it's `State::New` returns an error, which can't work for registration / linking.

Reply:
Ah, this is wrong then. The function below needs to be fixed.

Reply:
I have some improvement working on top of your PR that's also related, I'll add one commit on top that we can have a look at together.

Reply:
Perfect. Waiting for it.